x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/evaluation/regression/MeanSquaredError.java

```diff
@@ -48,30 +48,40 @@ public static MeanSquaredError fromXContent(XContentParser parser) {
         return PARSER.apply(parser, null);
     }
 
-    public MeanSquaredError(StreamInput in) {
+    private EvaluationMetricResult result;
 
-    }
-
-    public MeanSquaredError() {
+    public MeanSquaredError(StreamInput in) {}
 
-    }
+    public MeanSquaredError() {}
 
     @Override
-    public String getMetricName() {
+    public String getName() {
         return NAME.getPreferredName();
     }
 
     @Override
     public List<AggregationBuilder> aggs(String actualField, String predictedField) {
-        return Collections.singletonList(AggregationBuilders.avg(AGG_NAME).script(new Script(buildScript(actualField, predictedField))));
+        if (result != null) {
+            return List.of();
+        }
+        return List.of(AggregationBuilders.avg(AGG_NAME).script(new Script(buildScript(actualField, predictedField))));
     }
 
     @Override
-    public EvaluationMetricResult evaluate(Aggregations aggs) {
+    public void process(Aggregations aggs) {
+        result = evaluate(aggs);
+    }
+
+    private EvaluationMetricResult evaluate(Aggregations aggs) {
```

Comment:
Do we need this method now? We could inline that in `process`, no?

Reply:
Done.